src/components/tabs/tabs.ts

```diff
@@ -0,0 +1,400 @@
+    }));
+
+    const isNext = boundary === 'end';
+
+    const point = this.isLTR
+      ? isNext
+        ? forward
+        : backward
+      : isNext
+      ? backward
+      : forward;
+
+    let target = 0;
+
+    if (this.isLTR) {
+      target = isNext
+        ? dimensions.findIndex(
+            ({ start, end }) => start <= point && end >= point
+          )
+        : findLastIndex(dimensions, ({ start }) => start <= point);
+    } else {
+      target = isNext
+        ? dimensions.findIndex(
+            ({ start, end }) => start <= point && end >= point
+          )
+        : findLastIndex(dimensions, ({ end }) => end >= point);
+    }
+    return this.tabs[target];
+  }
+
+  protected handleStartButtonClick() {
+    this.scrollToTab(this.getTabAtBoundary('start'), 'start');
+  }
+
+  protected handleEndButtonClick() {
+    this.scrollToTab(this.getTabAtBoundary('end'), 'end');
+  }
+
+  protected handleClick(event: MouseEvent) {
+    const target = event.target as HTMLElement;
+    const tab = target.closest('igc-tab');
+
+    if (!tab || tab.disabled) {
+      return;
+    }
+
+    tab.focus();
+    this.setSelectedTab(tab);
+    this.emitEvent('igcChange', { detail: this._selected });
+  }
+
+  protected handleKeyDown = (event: KeyboardEvent) => {
+    const { key } = event;
+    const enabledTabs = this.enabledTabs;
+
+    let index = enabledTabs.indexOf(document.activeElement as IgcTabComponent);
+
+    switch (key) {
+      case 'ArrowLeft':
+        index = this.isLTR
+          ? (enabledTabs.length + index - 1) % enabledTabs.length
+          : (index + 1) % enabledTabs.length;
+        break;
+      case 'ArrowRight':
+        index = this.isLTR
+          ? (index + 1) % enabledTabs.length
+          : (enabledTabs.length + index - 1) % enabledTabs.length;
+        break;
+      case 'Home':
+        index = 0;
+        break;
+      case 'End':
+        index = enabledTabs.length - 1;
+        break;
+      case 'Enter':
+      case ' ':
+        this.setSelectedTab(enabledTabs[index]);
+        break;
+      default:
+        return;
+    }
+
+    enabledTabs[index].focus({ preventScroll: true });
+
+    if (this.activation === 'auto') {
+      this.setSelectedTab(enabledTabs[index]);
+      this.emitEvent('igcChange', { detail: this._selected });
+    } else {
+      this.scrollToTab(enabledTabs[index]);
+    }
+
+    event.preventDefault();
+  };
+
+  @eventOptions({ passive: true })
+  protected handleScroll() {
+    this.updateScrollButtons();
+  }
+
+  protected handleSlotChange = () => {
+    this.resizeObserver?.disconnect();
+
+    if (!this.selectedTab) {
+      this.tabs.forEach((tab) => (tab.selected = false));
+      this.panels.forEach((panel) => (panel.selected = false));
+      this._selected = '';
+    }
+
+    this.setupObserver();
+  };
+
+  /** Selects the specified tab and displays the corresponding panel.  */
+  public select(selectedTab: string | HTMLElement) {
```

Comment:
AFAIK the **selected** property does not notify the parent about the state change.
Otherwise, I'm ok with the method constrained to just the id of the tab (that is a string).

Reply:
Oh, well perhaps it should? I mean it's public API and you can set it, right?